### PR TITLE
Toolbar and ToolbarButton Updated

### PR DESCRIPTION
**Pull request by Kamianz**
ToolbarButton and toolbar are related to each other.

It get the colors automatically and we can change them manually.

### CODE REVIEW

**Comment by noyanc** [comment]
Why we need that update in colors? What does this PR do?

**Comment by irrld** [comment]
Why do you store an array of pointers for colors (gColor**) when it's just used as a single length array? Perhaps you misunderstood the pointer logic? Colors are already stored as pointers, you don't need to get their address, it's unnecessary. You don't need to name the color variables as toolbarforegroundcoloaradress.

File: engine/ui/gGUIToolbar.cpp [hunk 1]
```diff
@@ -12,8 +12,8 @@ gGUIToolbar::gGUIToolbar() {
 	spaceLocation = 0;
 	isSpaceAdded = false;
 	toolbartype = TOOLBAR_HORIZONTAL;
-	toolbarforegroundcolor = foregroundcolor;
-	toolbarbottomlinecolor = backgroundcolor;
+	toolbarforegroundcoloaradress = &foregroundcolor;
```
Comment (irrld) [inline engine/ui/gGUIToolbar.cpp:15]: This is unnecessary.

File: engine/ui/gGUIToolbar.cpp [hunk 2]
```diff
@@ -28,19 +28,20 @@ int gGUIToolbar::getToolbarType() {
 }
 
 void gGUIToolbar::draw() {
+	// The foregroundcolor and backgroundcolor assignments have been made.
 //	gLogi("gGUIToolbar") << "draw";
 //	gLogi("gGUIToolbar") << "l:" << left << ", t:" << top << ", w:" << width << ", h:" << height;
 	gColor* oldcolor = renderer->getColor();
 	if(toolbartype == TOOLBAR_HORIZONTAL) {
-		renderer->setColor(toolbarforegroundcolor);
+		renderer->setColor(*toolbarforegroundcoloaradress);
```
Comment (irrld) [inline engine/ui/gGUIToolbar.cpp:36]: This is also unnecessary, you can make toolbarforegroundcolor gColor* and remove the dereference operation (*)

**Comment by irrld** [comment]
There are still few variables in gFont that uses camelCase for variable names.

**Comment by noyanc** [comment]
Is the visual quality of the font letters good when scaling the letters up?

**Pull request closed without merging**